scripts/spack/packages/conduit/package.py

```diff
@@ -407,35 +421,36 @@ def hostconfig(self):
 
         # extra fun for blueos
         if on_blueos:
-            # All of BlueOS compilers report clang due to nvcc,
-            # override to proper compiler family
-            if "xlc" in c_compiler:
-                cfg.write(cmake_cache_entry("CMAKE_C_COMPILER_ID", "XL"))
-            if "xlC" in cpp_compiler:
-                cfg.write(cmake_cache_entry("CMAKE_CXX_COMPILER_ID", "XL"))
+
+            # AXOM EDIT START
+            # Don't override compiler family!
+            ## All of BlueOS compilers report clang due to nvcc,
+            ## override to proper compiler family
+            #if "xlc" in c_compiler:
+            #    cfg.write(cmake_cache_entry("CMAKE_C_COMPILER_ID", "XL"))
+            #if "xlC" in cpp_compiler:
+            #    cfg.write(cmake_cache_entry("CMAKE_CXX_COMPILER_ID", "XL"))
+            # AXOM EDIT END
 
             if "+fortran" in spec:
-                if "xlf" in f_compiler:
-                    cfg.write(cmake_cache_entry("CMAKE_Fortran_COMPILER_ID",
-                                                "XL"))
+                # AXOM EDIT START
+                #if "xlf" in f_compiler:
+                #    cfg.write(cmake_cache_entry("CMAKE_Fortran_COMPILER_ID", "XL"))
+                # AXOM EDIT START
```

Comment:
I had to comment these out since the `c++11` flags were not getting added for "XL" on blueos, but were getting set without it.

It looks like there's a bug in BLT related to attempting to set custom compiler flags before establishing the compiler families. I'll open a ticket there.

See:
* https://github.com/LLNL/blt/blob/5644d546bbc168020e8c6b8e63987ad52ae9a9d1/SetupBLT.cmake#L122-L211
* https://github.com/LLNL/blt/blob/develop/cmake/SetupCompilerStandards.cmake
* https://github.com/LLNL/blt/blob/develop/cmake/SetupCompilerOptions.cmake

Tag: @cyrush